workers/unity/Assets/Fps/Scripts/UI/Controllers/ConnectScreenController.cs

```diff
@@ -0,0 +1,35 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+
+namespace Fps
+{
+    public class ConnectScreenController : MonoBehaviour
+    {
+        public Button ConnectButton;
+
+        public void Awake()
+        {
+            ConnectButton.onClick.AddListener(ConnectClicked);
+        }
+
+        public void ConnectClicked()
+        {
+            if (ConnectionStateReporter.IsConnected)
+            {
+                ConnectionStateReporter.TrySpawn();
```

Comment:
Why does a connection button change functionality if you are already connected?

Reply:
I believe this is just the way the default screen is set up - one button does everything.

Reply:
To clarify, "It was like that when I got here, 'guv".